aws-samples/aws-tools-for-babylonjs-editor
# Credential management refactor and Host script improvements

## Issue #, if available:

(AWS internal) 11365, 11366

## Description of changes:

- Renamed all Editor scripts to end with suffix "Script".
- Changed `SumerianHostScript.host` property from _private_ to _public_ to allow access to the full Host API from other scripts.
- Extracted credential management to new **AwsCognitoIdConnectorScript** class to separate concerns, allowing multiple hosts and Lex chatbot scripts to share the same credentials.
- Added new "HelloWorldScript.ts" script to ease developer onboarding and testing.
- Any files contained within the host plugin's `scripts/AwsTools/` folder will be copied to `<workspace>/src/scenes/AwsTools/` when adding a host to a workspace.
- **SumerianHostScript** now exposes inspectable parameters for "Voice ID", "Language ID", and "Polly Engine" enabling developers to easily customize these values per instance and without editing code. ✏️ **Note:** As a side effect of this change, all hosts default to the voice "Joanna" regardless of their gendered appearance. This is a limitation of the Babylon.JS Editor's inspectable property mechanism.
- Updated and expanded hosts plugin README documentation.
- Made unit tests more robust.
- Fixed file path compatibility problems on Windows.

## How to test:

Please be sure you're using v4.2.0 of the Editor. Our Hosts library is momentarily incompatible with the just-released v4.3.0 Editor.

Thanks to the new "HelloWorldScript.ts", creating a fully functioning scene to test behavior is pretty easy. Just follow the instructions in the README.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
